### PR TITLE
Convert navbar to use Flask-Copilot

### LINKED ISSUE #219: Should we use Flask-Menu?
[Flask-Menu](http://flask-menu.readthedocs.org) is probably a simpler way to handle the changes introduced in 0418bda to address site navigation.

### PR DESCRIPTION
**Pull request by jonafato**
Replace custom and inflexible navbar generation code with Flask-Copilot.
Although this makes adjusting placement of navbar content easier, no
items are moved in this change (i.e. this is a strictly transparent port
with no desired functional side effects).

Fixes #219.

### CODE REVIEW

File: pygotham/frontend/sponsors.py [hunk 1]
```diff
@@ -73,7 +73,7 @@ def edit(pk):
     return render_template('sponsors/edit.html', form=form, sponsor=sponsor)
 
 
-@route(blueprint, '/')
+@route(blueprint, '/', navbar_kwargs={'path': ('Sponsors', 'Sponsors')})
```
Comment (dirn) [inline pygotham/frontend/sponsors.py]: Can this be reduced to just the Sponsors top nav item as per our discussions with Allison?

File: pygotham/frontend/talks.py [hunk 2]
```diff
@@ -60,7 +65,11 @@ def index():
     '/new/',
     defaults={'pk': None},
     endpoint='submit',
-    methods=('GET', 'POST'))
+    methods=('GET', 'POST'),
+    navbar_kwargs={
+        'path': ('Speaking', 'Submit a talk'),
+        'when': lambda: g.current_event.is_call_for_proposals_active,
```
Comment (dirn) [inline pygotham/frontend/talks.py]: Unrelated, but we should probably rename this to `call_for_proposals_is_active` to be consistent with the others.

File: pygotham/frontend/talks.py [hunk 3]
```diff
@@ -60,7 +65,11 @@ def index():
     '/new/',
     defaults={'pk': None},
     endpoint='submit',
-    methods=('GET', 'POST'))
+    methods=('GET', 'POST'),
+    navbar_kwargs={
+        'path': ('Speaking', 'Submit a talk'),
```
Comment (dirn) [inline pygotham/frontend/talks.py]: This used to be "Submit a Talk." Was the case change intentional?

File: pygotham/frontend/sponsors.py (hunk 1 above)
Comment (jonafato) [inline pygotham/frontend/sponsors.py]: I definitely think this should be renamed, but I think that's out of scope of this PR. The renaming should probably happen in one swoop once we have our new navbar hierarchy laid out.

Comment (dirn) [inline pygotham/frontend/sponsors.py]: I wasn't saying that we should do it as part of this. I was just asking if Flask-Copilot allowed us to do so. I'm guessing that's where `default_href` comes into play, but I just wanted to check.

File: pygotham/frontend/talks.py (hunk 3 above)
Comment (jonafato) [inline pygotham/frontend/talks.py]: This change was unintentional. Fixed.

File: pygotham/frontend/sponsors.py (hunk 1 above)
Comment (jonafato) [inline pygotham/frontend/sponsors.py]: Ah, yes, if we provide only one item to ``path`` (or just set it to the string), the top tier "Sponsors" item will become the link. Flask-Copilot uses `default_href` as a fallback when a navbar item has no endpoint assigned to it, so all of the top tier links with default to `#` right now.

**Pull request merged**